utils/cloud_cleanup/azure/clean_azure.py

```diff
@@ -0,0 +1,111 @@
+#
+# See LICENSE for more details.
+#
+# Copyright (c) 2024 ScyllaDB
+
+import argparse
+import os
+from datetime import datetime, timedelta
+
+from sdcm.utils.azure_utils import AzureService
+
+azure_service = AzureService()
+resource_client = azure_service.resource
+compute_client = azure_service.compute
+
+
+def get_keep_hours(v_m, default=14):
+    keep = v_m.tags.get('keep') if v_m.tags else None
+    if keep == "alive":
+        return -1
+    try:
+        return int(keep)
+    except (ValueError, TypeError):
+        return default
+
+
+def get_creation_time(v_m):
+    try:
+        creation_time = datetime.fromisoformat(v_m.tags.get('creation_time', "")) if v_m.tags else datetime.now()
+    except ValueError:
+        print("Error parsing creation time tag: %s" % v_m.tags.get('creation_time'))
+        # todo: set creation time to now in instance.tags so next time will be processed properly?
+        # (there's no easy way of getting creation time on azure)
+        # from the other hand, should not happen for SCT tests
+        creation_time = datetime.now()
+    return creation_time
+
+
+def should_keep_vm(creation_time, keep_hours):
+    if keep_hours <= 0:
+        return True
+    try:
+        keep_date = creation_time + timedelta(hours=keep_hours)
+        return datetime.now() < keep_date
+    except (TypeError, ValueError):
+        return False
+
+
+def delete_virtual_machine(resource_group_name, vm_name, dry_run=False):
+    if dry_run:
+        print(f"[DRY RUN] Would delete VM: {vm_name} in resource group: {resource_group_name}")
+    else:
+        print(f"Deleting VM: {vm_name} in resource group: {resource_group_name}")
+        compute_client.virtual_machines.begin_delete(resource_group_name, vm_name)
+
+
+def delete_resource_group(resource_group_name, dry_run=False):
+    if dry_run:
+        print(f"[DRY RUN] Would delete resource group: {resource_group_name}")
+    else:
+        print(f"Deleting resource group: {resource_group_name}")
+        resource_client.resource_groups.begin_delete(resource_group_name)
+
+
+def clean_azure_instances(dry_run=False):
+    groups = [group for group in resource_client.resource_groups.list() if group.name.startswith('SCT-')]
+    for group in groups:
+        print(f"Checking resource group: {group.name}")
+        clean_group = True
+        vms_to_delete = []
+        for v_m in compute_client.virtual_machines.list(group.name):
+            if should_keep_vm(creation_time=get_creation_time(v_m), keep_hours=get_keep_hours(v_m)):
+                print(f"Keeping VM: {v_m.name} in resource group: {group.name}")
+                clean_group = False
+            else:
+                vms_to_delete.append(v_m.name)
+
+        if clean_group:
+            delete_resource_group(group.name, dry_run=dry_run)
+        else:
+            for vm_name in vms_to_delete:
+                delete_virtual_machine(group.name, vm_name, dry_run=dry_run)
```

Comment:
I think it will be better to use `try`-block here to try to delete as much as possible resources.

Reply:
currently it just triggers delete (doesn't wait for the result), but anyway could fail for some reason.
I'll add it

Reply:
I think we should wait for result, maybe at the end of the loop ?